### PR TITLE
Implemented Plugin Metrics

### PR DESCRIPTION
**Pull request by SparklingComet**
At the moment, the bStats API classes were copied into the source. I will shade them into the JAR later on.

This also fixes a small Updater log issue.

### CODE REVIEW

File: src/main/java/org/shanerx/tradeshop/Metrics.java [hunk 1]
```diff
@@ -0,0 +1,688 @@
...
+ */
+public class Metrics {
+	
+	static {
+		// You can use the property to disable the check in your test environment
+		if (System.getProperty("bstats.relocatecheck") == null || !System.getProperty("bstats.relocatecheck").equals("false")) {
+			// Maven's Relocate is clever and changes strings, too. So we have to use this little "trick" ... :D
+			final String defaultPackage = new String(
+					new byte[]{'o', 'r', 'g', '.', 'b', 's', 't', 'a', 't', 's', '.', 'b', 'u', 'k', 'k', 'i', 't'});
+			final String examplePackage = new String(new byte[]{'y', 'o', 'u', 'r', '.', 'p', 'a', 'c', 'k', 'a', 'g', 'e'});
```
Comment (KillerOfPie) [inline src/main/java/org/shanerx/tradeshop/Metrics.java]: I would imagine that these should be changed and could be just normal strings, really dont know about that second one.

File: src/main/java/org/shanerx/tradeshop/Metrics.java [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+				throw new IllegalStateException("bStats Metrics class has not been relocated correctly!");
+			}
+		}
+	}
+	
+	// The version of this bStats class
+	public static final int B_STATS_VERSION = 1;
+	
+	// The url to which the data is sent
+	private static final String URL = "https://bStats.org/submitData/bukkit";
```
Comment (KillerOfPie) [inline src/main/java/org/shanerx/tradeshop/Metrics.java]: same with this

**Review by KillerOfPie — CHANGES_REQUESTED** [review]
I would assume the rest is fine but didn't look through all of metrics the other 2 are fine

File: src/main/java/org/shanerx/tradeshop/Metrics.java (hunk 1 above)
Comment (SparklingComet) [reply src/main/java/org/shanerx/tradeshop/Metrics.java]: Thought about it too. Not really sure what is going on. I just copy-pasted the class as the instructions told me.

Comment (SparklingComet) [reply src/main/java/org/shanerx/tradeshop/Metrics.java]: 2 tab indentation is also not really optimal. I will make sure to reformat the code.

**Pull request merged**